GTG/gtk/browser/taskbrowser.ui

```diff
@@ -8,529 +8,176 @@
-                    <property name="homogeneous">True</property>
-                  </packing>
+                  <style>
+                    <class name="text-button"/>
+                  </style>
                 </child>
                 <child>
-                  <object class="GtkSeparatorToolItem" id="separator_7">
+                  <object class="GtkToggleButton" id="closed_tasks">
                     <property name="visible">True</property>
-                    <property name="can_focus">False</property>
+                    <property name="label" translatable="yes">Closed Tasks</property>
+                    <property name="tooltip_text" translatable="yes">View Closed Tasks</property>
+                    <signal name="clicked" handler="on_view_closed_toggled" swapped="no"/>
+                    <property name="valign">center</property>
                   </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="homogeneous">True</property>
-                  </packing>
+                  <style>
+                    <class name="text-button"/>
+                  </style>
                 </child>
+              </object>
+            </child>
+            <child>
+              <object class="GtkButton" id="search">
+                <property name="visible">True</property>
+                <property name="sensitive">True</property>
+                <property name="tooltip_text" translatable="yes">Searching thing</property>
+                <signal name="clicked" handler="temporary_search" swapped="no"/>
+                <property name="valign">center</property>
                 <child>
-                  <object class="GtkToggleToolButton" id="workview_toggle">
+                  <object class="GtkImage" id="search_icon">
                     <property name="visible">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="is_important">True</property>
-                    <property name="label" translatable="yes">Work View</property>
-                    <property name="stock_id">gtk-index</property>
-                    <signal name="toggled" handler="on_workview_toggled" swapped="no"/>
+                    <property name="icon-name">edit-find-symbolic</property>
+                    <property name="icon-size">1</property>
                   </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="homogeneous">True</property>
-                  </packing>
                 </child>
               </object>
               <packing>
-                <property name="expand">False</property>
-                <property name="fill">True</property>
-                <property name="position">0</property>
+                <property name="pack_type">end</property>
```

Comment:
I don't think that you intended to put the search_icon to the very right side of the pane ;)